Feature/1200453899422605/expand date filters

jrrlokken opened the pull request:
## Description

Added additional DatePicker components and logic to append all required dates to db query, allowing for filtering organizations by date ranges for last_verified, last_updated and created_at dates.

## Asana ticket:  https://app.asana.com/0/1132189118126148/1200453899422605

## PR Checklist

<!-- Please validate your changes with the checklist below before marking for code review. -->

- [x] Assign @FJKhan **and** @Alfredo-Moreira as reviewers.
- [x] If your PR is not a hotfix, is it targeted for `dev`? If your PR is a hotfix, is it targeted for `master`?
- [ ] Unit and functional test coverage was added where applicable.
- [x] CI/CD passes for your PR.
- [ ] Complex code is well documented with comments.
- [x] Does the original ticket have test instructions? If not add them below

## How to Test

1. Login to Control Panel.
2. Under Filter Organizations, enter a date or use the date picker to populate either the first date field or both date fields under Last Verified, Last Updated or Created At
3. Click Search
4. Click View next to one of the returned results to view the organization
5. If only the first date field was used, verify that the returned results have a [Last Verified, Last Updated or Created At] dates older than the date entered. If both date fields for a criteria were used, verify that the returned results have a [Last Verified, Last Updated or Created At] date between the date values entered.

FJKhan commented on src/date-picker.css:
```diff
@@ -9,10 +9,12 @@
+  font-size: 0.8rem;
+  padding-left: 0.5rem;
+  padding-right: 0.5rem;
+  /* margin: 0 0 0 2rem; */
```
can you remove the lines instead of commenting them out, please?

jrrlokken replied:
You bet -- done.

FJKhan commented on src/components/FiltersOrganizations.js:
```diff
@@ -185,11 +207,23 @@ const FiltersOrganizations = (props) => {
+        <DateFieldPicker selected={createdAt} onChange={setCreatedAt} />
+        <DateFieldPicker selected={createdAtEnd} onChange={setCreatedAtEnd} />
 
         <Text>Created Before:</Text>
```
remember to remove this previous filter.

jrrlokken replied:
Extraneous date filter removed.

FJKhan reviewed (changes requested):
can we get better labels on the filters please? I can't tell which is for "before" and "after".
also, there are now 3 filters related to the "Created at" date. two of which are I'm assuming for "created before".

![image](https://user-images.githubusercontent.com/7406914/124621421-41ed1480-de48-11eb-9440-5f46515dcaaa.png)

jrrlokken commented:
Improved labeling/layout of date filters:

![1200453899422605](https://user-images.githubusercontent.com/50845333/124813480-8bae2b80-df2a-11eb-8bb4-3ca91434281d.gif)

FJKhan reviewed (changes requested):
@jrrlokken thanks for the updates! this is looking so much better! Amazing job, bud!

a few things I noticed:
 - the placeholder text for using a single date vs date range are different. I think both should say "select x date"
 
![image](https://user-images.githubusercontent.com/7406914/125808593-1624ece4-1cfc-48a1-8e16-1103f407ed6b.png)

- when I switch from using a date range to a single date, the second (end) date value is retained which leads to incorrect search results
 
![image](https://user-images.githubusercontent.com/7406914/125808445-fb061617-12da-4ef2-8fba-80f138ef4fd1.png)

jrrlokken commented:
Addressed concerns about placeholder text, it is much simpler now, single date picker shows 'Select a date', date range pickers show 'Select start date' and 'Select end date'.  Validations have been implemented to prevent leaving behind stale dates and to prevent invalid or out-of-scope dates.
